### PR TITLE
Adding ability to display static entity widget in launcher

### PR DESCRIPTION
**Pull request by colincachia**
Hi, I would personally find this feature interesting. Displaying some important entity state information on the launcher homescreen would avoid me a couple of seconds every time to get into the home assistant app.

### CODE REVIEW

**Comment by homeassistant** [comment]
Hi @colincachia, 

It seems you haven't yet signed a CLA. Please do so [here](https://home-assistant.io/developers/cla_sign_start/?pr=home-assistant&#x2F;android#687).

Once you do that we will be able to review and accept this pull request.

Thanks!

**Comment by colincachia** [comment]
I guess there are some merge conflicts due to a recent update to translation files. Is there anything I should do in this PR to update any newly added fields?

**Comment by JBassett** [comment]
Also sorry for not getting to this sooner!

**Comment by colincachia** [comment]
> I know we have talked about adding this in the past. The worry was that the data displayed would not be up to date with changes. I think some of that can be alleviated by possibly adding a refresh icon or just adding an on click action to the whole view that refreshes the widget? We can expand upon this in the future to support templating as well. That way you could write a jinja2 template and have it rendered which I think would be great!

Having an on click action is the easiest way to go - but also the more efficient. Another approach could be to have an update interval specified in the widget setup. I can add either one :) kindly let me know your preference

File: app/src/main/java/io/homeassistant/companion/android/widgets/StaticWidgetConfigureActivity.kt [hunk 1]
```diff
@@ -0,0 +1,192 @@
...
+
+        mainScope.launch {
+            try {
+                // Fetch entities
+                val fetchedEntities = integrationUseCase.getEntities()
+                fetchedEntities.sortBy({ e -> e.entityId })
+                fetchedEntities.forEach {
+                    entities[it.entityId] = it
+                }
+                i(TAG, "Successfully queried entities [" + fetchedEntities + ".")
```
[the remark this answers is not in the document]
Comment (colincachia) [reply app/src/main/java/io/homeassistant/companion/android/widgets/StaticWidgetConfigureActivity.kt:104]: Agreed, will remove as it gives little value

File: app/src/main/java/io/homeassistant/companion/android/widgets/StaticWidgetConfigureActivity.kt [hunk 2]
```diff
@@ -0,0 +1,192 @@
...
+                entityAdapter.addAll(entities.values)
+                entityAdapter.sort()
+
+                runOnUiThread {
+                    entityAdapter.notifyDataSetChanged()
+                }
+            } catch (e: Exception) {
+                // If entities fail to load, it's okay to pass
+                // an empty map to the dynamicFieldAdapter
+                Log.e(TAG, "", e)
```
[the remark this answers is not in the document]
Comment (colincachia) [reply app/src/main/java/io/homeassistant/companion/android/widgets/StaticWidgetConfigureActivity.kt:114]: Apologies, will do

File: app/src/main/java/io/homeassistant/companion/android/widgets/StaticWidgetConfigureActivity.kt [hunk 3]
```diff
@@ -0,0 +1,192 @@
...
+        val fetchedAttributes = selectedEntity?.attributes as Map<String, String>
+        val attributesAdapter = SingleItemArrayAdapter<String>(this) { it ?: "" }
+        widget_text_config_attribute.setAdapter(attributesAdapter)
+        attributesAdapter.addAll(*fetchedAttributes.keys.toTypedArray())
+        runOnUiThread {
+            attributesAdapter.notifyDataSetChanged()
+        }
+    }
+
+    private var onClickListener = View.OnClickListener {
```
[the remark this answers is not in the document]
Comment (colincachia) [reply app/src/main/java/io/homeassistant/companion/android/widgets/StaticWidgetConfigureActivity.kt:146]: My bad, will rename to something more descriptive

File: app/src/main/res/layout/widget_static.xml [hunk 4]
```diff
@@ -0,0 +1,59 @@
...
+
+    <ProgressBar
+        android:id="@+id/widgetProgressBar"
+        android:layout_width="match_parent"
+        android:layout_height="match_parent"
+        android:padding="8dp"
+        android:visibility="invisible"
+        android:indeterminate="true"
+        android:indeterminateTint="@color/colorAccent"
+        />
```
[the remark this answers is not in the document]
Comment (colincachia) [reply app/src/main/res/layout/widget_static.xml:58]: Whoops, will remove 😃

**Comment by colincachia** [comment]
> Also sorry for not getting to this sooner!

No worries, thanks for the review! Will implement the suggested changes later on today or tomorrow 😃

**Comment by JBassett** [comment]
> Another approach could be to have an update interval specified in the widget setup.

I think the minimum is 30 minutes for `updatePeriodMillis` just as a warning.

**Comment by colincachia** [comment]
> 
> 
> > Another approach could be to have an update interval specified in the widget setup.
> 
> I think the minimum is 30 minutes for `updatePeriodMillis` just as a warning.

You're right, any value less than 30 minutes will be ignored by android. I was thinking more on the lines of somehow running a TimerTask alongside the widget, although I'm not 100% sure that it is possible.

**Pull request merged**